packages/core/src/animation/AnimatorTransition.ts

```diff
@@ -12,4 +15,79 @@ export class AnimatorStateTransition {
   exitTime: number = 1;
   /** The destination state of the transition. */
   destinationState: AnimatorState;
+  /** Mutes the transition. The transition will never occur. */
+  mute: boolean = false;
+  /** Is the transition destination the exit of the current state machine. */
+  isExit: boolean = false;
+  /** @internal */
+  _srcState: AnimatorState;
+
+  private _conditions: AnimatorCondition[] = [];
+  private _solo = false;
+
+  /** Mutes all other transitions in the source state. */
+  get solo(): boolean {
+    return this._solo;
+  }
+
+  set solo(value: boolean) {
+    if (this._solo === value) return;
+    this._solo = value;
+    if (value) {
+      this._srcState && this._srcState._updateSoloTransition(true);
+    } else {
+      this._srcState && this._srcState._updateSoloTransition();
+    }
+  }
+  /**
+   * The conditions in the transition.
+   */
+  get conditions(): Readonly<AnimatorCondition[]> {
+    return this._conditions;
+  }
+
+  /**
+   * Add a condition to a transition.
+   * @param mode - The AnimatorCondition mode of the condition
+   * @param parameter - The name of the parameter
+   * @param threshold - The threshold value of the condition
+   */
+  addCondition(
+    mode: AnimatorConditionMode,
+    parameter: string,
+    threshold?: AnimatorControllerParameterValue
```

Comment:
-  Why threshold can be undefined
- The order should same with `AnimatorCondition` constructor

Reply:
Because AnimatorConditionMode may be  IF means the parameter is true, AnimatorCondition will add '?'